localstack-core/localstack/services/cloudwatch/provider_v2.py

```diff
@@ -445,21 +449,13 @@ def put_metric_alarm(self, context: RequestContext, request: PutMetricAlarmInput
 
     @handler("PutCompositeAlarm", expand=False)
     def put_composite_alarm(self, context: RequestContext, request: PutCompositeAlarmInput) -> None:
-        composite_to_metric_alarm = {
-            "AlarmName": request.get("AlarmName"),
-            "AlarmDescription": request.get("AlarmDescription"),
-            "AlarmActions": request.get("AlarmActions", []),
-            "OKActions": request.get("OKActions", []),
-            "InsufficientDataActions": request.get("InsufficientDataActions", []),
-            "ActionsEnabled": request.get("ActionsEnabled", True),
-            "AlarmRule": request.get("AlarmRule"),
-            "Tags": request.get("Tags", []),
-        }
-        self.put_metric_alarm(context=context, request=composite_to_metric_alarm)
-
-        LOG.warning(
-            "Composite Alarms configuration is not yet supported, alarm state will not be evaluated"
-        )
+        with _STORE_LOCK:
+            store = self.get_store(context.account_id, context.region)
+            composite_alarm = LocalStackCompositeAlarm(
+                context.account_id, context.region, {**request}
+            )
+            alarm_arn = composite_alarm.alarm["AlarmArn"]
```

Comment:
TODO: input validation, need to check whether alarms referenced in the rule exist in the store.

Reply:
added, see https://github.com/localstack/localstack/pull/11828#discussion_r1838458795